cni/plugin.go

```diff
@@ -197,26 +200,30 @@ func (plugin *Plugin) IsSafeToRemoveLock(processName string) (bool, error) {
 			return false, cmdErr
 		}
 
-		// Read pid from lockfile
 		lockFileName := plugin.Store.GetLockFileName()
-		content, err := ioutil.ReadFile(lockFileName)
+		// Read pid from lockfile
+		lockFilePid, err := plugin.readLockFile(lockFileName)
 		if err != nil {
-			log.Errorf("Failed to read lock file :%v, ", err)
-			return false, err
-		}
-
-		if len(content) <= 0 {
-			log.Errorf("Num bytes read from lock file is 0")
-			return false, fmt.Errorf("Num bytes read from lock file is 0")
+			return false, errors.Wrap(err, "IsSafeToRemoveLock lockfile read failed")
 		}
 
-		log.Printf("Read from Lock file:%s", content)
+		log.Printf("Read from Lock file:%s", lockFilePid)
 		// Get the process name if running and
 		// check if that matches with our expected process
-		pName, err := platform.GetProcessNameByID(string(content))
+		pName, err := platform.GetProcessNameByID(lockFilePid)
 		if err != nil {
```

Comment:
will add comment but i dont want to change logic now since i need to verify both linux and windows. anyway this code will go away once we move to filelock api.

Reply:
@ashvindeodhar I disagree. Semantically, `GetProcessNameByID` should only *not* return an error if and only if a process name is found and returned. If a process name in not found, it will return an error, because that's the contract implied by the method name, so the most correct thing to check is if we received an error or not.
This is pretty standard in Go and you can see examples of it in e.g. the controller-runtime client, where the `Get` method will return an error of type ErrNotFound